### PR TITLE
feature: make the scaling absolute across all the trials

### PR DESCRIPTION
**Pull request by gcroci2**
- The scaling in the input signals is done after the noise addition - no edit here, it was already like that. 
- `stim_intensities` is now used as a list of possible values for the stimulus, i.e., when the stimulus is present. The absence of the stimulus is encoded with a 0 value. Then, the noise is added and eventually, the signals are scaled between 0 and 1 (if `scaling` is `True`). I clarified this in the `Task`'s doc string. 
- Now the scaling is done after having created all input and output signals, and the absolute maximum and minimum values are used for doing it. So if `fix_intensity` is set to 2 and `stim_intensities` has lower values, and with the noise addition it becomes e.g. 2.15, the latter will be considered as the highest value in the trials and used to scale all the signals in the range 0-1. This way the relativity across signals is maintained, but everything is in the 0-1 range.

### CODE REVIEW

**Comment by gcroci2** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ANNUBS/annubes/90?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#90** <a href="https://app.graphite.dev/github/pr/ANNUBS/annubes/90?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#88** <a href="https://app.graphite.dev/github/pr/ANNUBS/annubes/88?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gcroci2 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: annubes/task.py [hunk 1]
```diff
@@ -434,30 +447,35 @@ def _setup_trial_phases(
...
     def _minmaxscaler(
         self,
-        input_: NDArray[np.float64],
+        array_: NDArray[np.float64],
+        abs_min: float,
+        abs_max: float,
         rescale_range: tuple[float, float] = (0, 1),
     ) -> NDArray[np.float64]:
-        """Rescale `input_` array to a given range.
+        """Rescale `array_` array to a given range.
```
Comment (DaniBodor) [inline annubes/task.py:455]:
```suggestion
        """Rescale `array_` to a given range.
```

**Review by DaniBodor — COMMENTED** [review]
How does this relate to trials from different epochs during training?
From memory, I think it was also important that the degree of rescaling does not change between epochs, even if that means that the resulting scale is not exactly (0,1) in every epoch. Or am I misremembering this?

**Comment by gcroci2** [comment]
> How does this relate to trials from different epochs during training? From memory, I think it was also important that the degree of rescaling does not change between epochs, even if that means that the resulting scale is not exactly (0,1) in every epoch. Or am I misremembering this?

An epoch is a complete pass of the entire dataset (for us, all the generated trials, being `trials` our dataset), so these edits shouldn't influence that part at all.

**Review by DaniBodor — APPROVED** [review]

**Pull request merged**